pulsar-broker/src/main/java/org/apache/pulsar/broker/service/BrokerService.java

```diff
@@ -691,8 +693,8 @@ private void createPersistentTopic(final String topic, boolean createIfMissing,
                         @Override
                         public void openLedgerComplete(ManagedLedger ledger, Object ctx) {
                             try {
-                                PersistentTopic persistentTopic = new PersistentTopic(topic, ledger,
-                                        BrokerService.this);
+                                PersistentTopic persistentTopic = new PersistentTopic(
```

Comment:
I am actually wondering if it is actually clearer if we create a class `SystemTopic` extending `PersistentTopic`. So it will be:

```
class SystemTopic extends PersistentTopic {
}

PersistentTopic persistentTopic;
if (isSystemTopic(topic)) {
    persistentTopic = new PersistentTopic(topic, ledger, BrokerService.this);
} else {
    persistentTopic = new SystemTopic(topic, ledger, BrokerService.this);
}
```